### PR TITLE
docs: Update some docs around Helm Values changes 

### PR DESCRIPTION
**Pull request by jonathan-innis**
Related to #4905 

**Description**

- Add an update to the upgrade reference about helm values changes in v1beta1
- Removes all deprecated values from the current `values.yaml` file in the chart to avoid confusion

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/karpenter/README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ helm upgrade --install --namespace karpenter --create-namespace \
...
+| settings | object | `{"assumeRoleARN":"","assumeRoleDuration":"15m","aws":{"assumeRoleARN":"","assumeRoleDuration":"15m","clusterCABundle":"","clusterEndpoint":"","clusterName":"","defaultInstanceProfile":"","enableENILimitedPodDensity":true,"enablePodENI":false,"interruptionQueueName":"","isolatedVPC":false,"reservedENIs":"0","tags":null,"vmMemoryOverheadPercent":0.075},"batchIdleDuration":"1s","batchMaxDuration":"10s","clusterCABundle":"","clusterEndpoint":"","clusterName":"","featureGates":{"drift":false,"driftEnabled":false},"interruptionQueue":"","isolatedVPC":false,"reservedENIs":"0","vmMemoryOverheadPercent":0.075}` | Global Settings to configure Karpenter |
```
Comment (njtran) [inline charts/karpenter/README.md]: Doesn't this get overridden anyway?

File: website/content/en/preview/upgrading/v1beta1-reference.md [hunk 2]
```diff
@@ -604,23 +604,49 @@ In addition to these metrics, the MachineNotFound error returned by the `karpent
...
+| vmMemoryOverheadPercent                       | VM_MEMORY_OVERHEAD_PERCENT | --vm-memory-overhead-percent |
+| interruptionQueueName                         | INTERRUPTION_QUEUE         | --interruption-queue     |
+| reservedENIs                                  | RESERVED_ENIS              | --reserved-enis          |
+| featureGates.driftEnabled                     | FEATURE_GATE="Drift=true"  | --feature-gates Drift=true |
+
+## Helm Values
+
+The v1beta1 helm chart comes with a number of changes to the values that were previously used in v0.31.x. Your older helm values will continue to work throughout v0.32.x but any values no longer specified in the chart will no longer be supported starting in v0.33.0.
+
+| < v0.32.x Key                           | >= v0.32.x Key                   |
```
Comment (njtran) [inline website/content/en/preview/upgrading/v1beta1-reference.md:629]: thoughts on this?
```suggestion
|  Before v0.32.x Key                           | v0.32.x+ Key                   |
```

File: website/content/en/preview/upgrading/v1beta1-reference.md [hunk 3]
```diff
@@ -604,23 +604,49 @@ In addition to these metrics, the MachineNotFound error returned by the `karpent
...
+| settings.aws.clusterCABundle            | settings.clusterCABundle         |
+| settings.aws.clusterName                | settings.clusterName             |
+| settings.aws.clusterEndpoint            | settings.clusterEndpoint         |
+| settings.aws.defaultInstanceProfile     | Dropped                          |
+| settings.aws.enablePodENI               | Dropped                          |
+| settings.aws.enableENILimitedPodDensity | Dropped                          |
+| settings.aws.isolatedVPC                | settings.isolatedVPC             |
+| settings.aws.vmMemoryOverheadPercent    | settings.vmMemoryOverheadPercent |
+| settings.aws.interruptionQueueName      | settings.interruptionQueue       |
+| settings.aws.tags                       | Dropped                          |
```
Comment (njtran) [inline website/content/en/preview/upgrading/v1beta1-reference.md:647]: Can we make it just super clear that this is dropped? Maybe we can bring these to the top or the bottom together? 
```suggestion
| settings.aws.defaultInstanceProfile     | **Dropped**                          |
| settings.aws.enablePodENI               | **Dropped**                          |
| settings.aws.enableENILimitedPodDensity | **Dropped**                          |
| settings.aws.isolatedVPC                | settings.isolatedVPC             |
| settings.aws.vmMemoryOverheadPercent    | settings.vmMemoryOverheadPercent |
| settings.aws.interruptionQueueName      | settings.interruptionQueue       |
| settings.aws.tags                       | **Dropped**                          |
```

File: website/content/en/preview/upgrading/v1beta1-reference.md (hunk 2 above)
Comment (jonathan-innis) [reply website/content/en/preview/upgrading/v1beta1-reference.md:629]: I personally like the way that it is right now.

File: website/content/en/preview/upgrading/v1beta1-reference.md (hunk 3 above)
Comment (jonathan-innis) [reply website/content/en/preview/upgrading/v1beta1-reference.md]: Good callout, I moved all the dropped metrics to the bottom of their tables to make it more clear

**Review by njtran — APPROVED** [review]

**Pull request merged**